### PR TITLE
Api 9224 - API versions OAS in review

### PR DESCRIPTION
**Pull request by ediiotero**
Story: [API-9224](https://vajira.max.gov/browse/API-9224)

New endpoint `/docserver/versions/search?oas-status={status}`

This will return a list of API versions with an OAS status
 StatusEnum (inactive, active, review, error)

example: 
`/docserver/versions/search?oas-status=review`
`{
    "majorVersion": 1,
    "status": "review",
    "name": "One Ring API"
  }`

### CODE REVIEW

**Comment by mheusgit** [comment]
Everything appears to be good.

**Review by mheusgit — DISMISSED** [review]

File: src/api/version/apiVersion.service.ts [hunk 1]
```diff
@@ -143,6 +144,23 @@ class APIVersionService {
...
+  public async getAPIVersionsOASStatus(status: string): Promise<APIStatus[]> {
+    try {
+      const qb = this.entityManager.createQueryBuilder(
+        APIVersionModel,
+        'version',
+      );
+
+      qb.select(['spec.name', 'spec.status', 'version.major_version'], true)
+        .join('version.openAPISpecs', 'spec')
+        .where({ 'spec.status': status });
```
Comment (christineyost) [inline src/api/version/apiVersion.service.ts:157]: Just curious why the query builder was used here instead of the entity manager find functionality? It might be nice to use one or the other consistently throughout the project if possible

File: src/api/version/apiVersion.service.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-import APIVersion from './apiVersion.interface';
+import { APIVersion, APIStatus } from './apiVersion.interface';
 
 class APIVersionService {
   public client: AxiosInstance;
   public entityManager: EntityManager;
   constructor(orm: MikroORM) {
     this.client = axios.create();
-    this.entityManager = orm.em;
+    this.entityManager = (orm.em as unknown) as EntityManager;
```
Comment (christineyost) [inline src/api/version/apiVersion.service.ts:16]: Why was this change needed?

Comment (ediiotero) [reply src/api/version/apiVersion.service.ts:16]: this was needed to get access to the query builder

File: src/api/version/apiVersion.service.ts (hunk 1 above)
Comment (ediiotero) [reply src/api/version/apiVersion.service.ts:157]: I wasn't sure how to join tables and cleanly get the data I needed using find. Its prob me not being familiar with Mikro-orm. Happy to change this to use find if we want to stick to that. Maybe we can pair on it 🙂

File: test/api/version/apiVersion.controller.integration.ts [hunk 3]
```diff
@@ -150,6 +157,27 @@ describe('APIVersions Controller', () => {
...
+        await request
+          .post('/docserver/apis/one_ring/versions/0/oas')
+          .auth(process.env['ADMIN_USERNAME'], process.env['ADMIN_PASSWORD']);
+
+        const response = await request.get(
+          '/docserver/versions/search?oas-status=review',
+        );
+        expect(response.status).toEqual(200);
+        expect(response.body).toEqual(statusResponse);
+      });
```
[the remark this answers is not in the document]
Comment (ediiotero) [reply test/api/version/apiVersion.controller.integration.ts:172]: yup, they are wiped in the beforeEach

File: src/api/version/apiVersion.controller.ts [hunk 4]
```diff
@@ -150,6 +152,30 @@ class APIVersionController {
...
+      const status = reqQuery['oas-status'];
+
+      if (status && status.toUpperCase() in StatusEnum) {
+        const results = await this.apiVersionService.getAPIVersionsOASStatus(
+          status,
+        );
+        response.status(200).json(results);
+      }
+
+      response.sendStatus(422);
```
[the remark this answers is not in the document]
Comment (ediiotero) [reply src/api/version/apiVersion.controller.ts]: done

File: test/api/version/apiVersion.controller.integration.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ediiotero) [reply test/api/version/apiVersion.controller.integration.ts:171]: yes, its wiped in the beforeEach

**Review by dsteeve — APPROVED** [review]
looks good!
